sdk/metricsadvisor/perf-tests/ai-metrics-advisor/test/rootCauses.spec.ts

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import { PerfStressOptionDictionary, getEnvVar } from "@azure/test-utils-perfstress";
+import { MetricsAdvisorTest } from "./metricsAdvisor.spec";
+interface MetricsAdvisorTestOptions {}
+
+export class RootCauseTest extends MetricsAdvisorTest<MetricsAdvisorTestOptions> {
+  detectionConfigId: string;
+  incidentId: string;
+  public options: PerfStressOptionDictionary<MetricsAdvisorTestOptions> = {};
+  constructor() {
+    super();
+    this.detectionConfigId = getEnvVar("METRICS_ADVISOR_DETECTION_CONFIG_ID");
+    this.incidentId = getEnvVar("METRICS_ADVISOR_INCIDENT_ID");
+  }
+
+  async runAsync(): Promise<void> {
+    await this.client.getIncidentRootCauses(this.detectionConfigId, this.incidentId);
```

Comment:
Can the root causes be iterated over like the other two tests?

Reply:
No these can't be iterated over like the other responses. For the list methods we get paginated responses, so these are designed differently.

Reply:
Can you still iterate over the response items in some way?  The concern is if you don't iterate, then it's possible the SDK could optimize the call too much and we're not testing a real-world scenario.

Reply:
Here's an example from storage.  The storage example is a paged response so this might not apply to `getIncidentRootCauses()`, but would the `for` loop apply?

https://github.com/Azure/azure-sdk-for-js/blob/216f1835a243700a7693b197bd0214610a7c7209/sdk/storage/perf-tests/storage-blob/test/listBlobs.spec.ts#L32-L38

Reply:
Actually, in most cases our perf tests should match our samples.  Here's what I see for `getIncidentRootCauses()` in our samples:

https://github.com/Azure/azure-sdk-for-js/blob/e59290ba59b1239488a91cbead024970688b3651/sdk/metricsadvisor/ai-metrics-advisor/samples/typescript/src/incidentsAndAlerts.ts#L161-L162

So I would use the same code in this perf test (but the body of the `for` loop can be empty).